google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -283,6 +315,29 @@ public void testBuilder() {
     assertSame(ServiceOptions.getDefaultRetrySettings(), DEFAULT_OPTIONS.getRetrySettings());
   }
 
+  @Test
+  public void testBuilder_quotaProjectServiceOptionTakesPrecedence() {
+    TestServiceOptions noCredsWithQuotaProject =
+        TestServiceOptions.newBuilder()
+            .setCredentials(NoCredentials.getInstance())
+            .setQuotaProjectId("quota-project-id")
+            .build();
+    TestServiceOptions quotaProjectCredsWithQuotaProject =
+        TestServiceOptions.newBuilder()
+            .setCredentials(credentialsWithQuotaProject)
+            .setQuotaProjectId("quota-project-id")
+            .build();
```

Comment:
One more nit: can we test reversing the `.setCredentials` and `.setQuotaProjectId` just to be confident in our if set/override logic?

Reply:
done